.docker/Dockerfile

```diff
@@ -117,38 +117,46 @@ RUN echo "source ${USER_WORKSPACE}/install/setup.bash" >> /home/$USERNAME/.bashr
 FROM robot AS desktop
 
 ENV DEBIAN_FRONTEND=noninteractive
-ENV GZ_VERSION=garden
-
-# Install Gazebo Garden: https://gazebosim.org/docs/garden/install_ubuntu
-RUN sudo wget https://packages.osrfoundation.org/gazebo.gpg -O /usr/share/keyrings/pkgs-osrf-archive-keyring.gpg \
-    && echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/pkgs-osrf-archive-keyring.gpg] http://packages.osrfoundation.org/gazebo/ubuntu-stable $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/gazebo-stable.list > /dev/null \
-    && sudo apt-get -q update \
-    && sudo apt-get -y --quiet --no-install-recommends install \
-    gz-garden \
-    && sudo apt-get autoremove -y \
-    && sudo apt-get clean -y \
-    && sudo rm -rf /var/lib/apt/lists/*
+ENV GZ_VERSION=harmonic
+
+# Install Gazebo Harmonic: https://gazebosim.org/docs/harmonic/install_ubuntu
+USER root
```

Comment:
Might be good to add a comment here mentioning why we temporarily switch back to the `root` user

Reply:
Done.  I've been steered towards this by [hadolint](https://github.com/hadolint/hadolint).  Essentially, prefer "USER root" over sudo.  When chained commands run as both root and non-root, prefer 'gosu'

i.e. don't:


```
USER normal_user
RUN sudo cmd_as_root && \
    cmd_as_user && \
    sudo cmd_as_root
```

Instead

```
USER root
RUN cmd_as_root && \
    gosu normal_user cmd_as_user && \
    cmd_as_root
USER normal_user
```

Of course, this contradicts their best practice suggestion of minimizing user-switching with "USER" but (shrug)

Stylistically this could be changed throughout the Dockerfile but I wanted to keep the change surgical.

Reply:
Cool. I'm going to add an issue mentioning that this should get applied throughout the rest of the Dockerfile in the future